### PR TITLE
Videos should also be supported EpicDemo#25

### PR DESCRIPTION
**Pull request by diliger**
done.

### CODE REVIEW

File: src/Images/ViewModels/ConceptPage.json [hunk 1]
```diff
@@ -2,6 +2,8 @@
     "Html": "",
     "ImageURL$": "",
     "$ImageURL$": { "Bind": "URL" },
+    "ImageMimeType$": "",
+    "$ImageMimeType$": { "Bind": "MimeType" },
```
Comment (joozek78) [inline src/Images/ViewModels/ConceptPage.json]: that's deprecated. Please use `DefaultTemplate.ImageMimeType.Bind`

File: src/Images/ViewModels/ConceptPage.json.cs [hunk 2]
```diff
@@ -18,24 +18,32 @@ partial class ConceptPage : Page, IBound<Something> {
...
+        public string URL
+        {
+            get
+            {
+                return this.Data.Illustration?.Content?.URL;
+            }
+            set
+            {
+                if (URL != value)
+                    oldImageUrl = URL;
```
Comment (joozek78) [inline src/Images/ViewModels/ConceptPage.json.cs]: please include braces, always

Comment (joozek78) [inline src/Images/ViewModels/ConceptPage.json.cs]: `oldImageUrl` doesn't seem to be used

File: src/Images/ViewModels/ConceptPage.json.cs [hunk 3]
```diff
@@ -53,5 +61,23 @@ partial class ConceptPage : Page, IBound<Something> {
         void Handle(Input.Save action) {
             this.Transaction.Commit();
         }
+
+        Illustration InitIllustration() {
+            if (!string.IsNullOrEmpty(oldImageUrl)) {
```
Comment (joozek78) [inline src/Images/ViewModels/ConceptPage.json.cs]: please use current code style (braces in new line)

File: src/Images/ViewModels/ConceptPage.json.cs [hunk 4]
```diff
@@ -53,5 +61,23 @@ partial class ConceptPage : Page, IBound<Something> {
...
+                this.helper.DeleteFile(this.oldImageUrl);
+                oldImageUrl = string.Empty;
+            }
+
+            Illustration illustration = this.Data.Illustration;
+
+            if (illustration == null) {
+                illustration = new Illustration { Concept = this.Data };
```
Comment (joozek78) [inline src/Images/ViewModels/ConceptPage.json.cs]: shouldn't you also assign to `this.Data.Illustration`?

File: src/Images/ViewModels/IllustrationPage.json [hunk 5]
```diff
@@ -2,10 +2,14 @@
     "Html": "/Images/viewmodels/ImagesPage.html",
     "Name$": "",
     "$Name$": { "Bind": "Concept.Name" },
+    "MimeType$": "",
+    "$MimeType$": { "Bind": "Content.MimeType" },
```
Comment (joozek78) [inline src/Images/ViewModels/IllustrationPage.json]: please use `DefaultTemplate`

File: src/Images/wwwroot/Images/elements/so-juicy-filedrop/so-juicy-filedrop.html [hunk 6]
```diff
@@ -196,13 +231,17 @@
...
-                  //this caused https://github.com/StarcounterSamples/Images/issues/23
-                  this.openFile();
+                    //in IE and Edge, openClickChanged gets incorrectly called on page load, changing openClick value from "undefined" to 0
+                    //this caused https://github.com/StarcounterSamples/Images/issues/23
+                    this.openFile();
                 }
             },
             getStyle: function (url) {
-                return "background-image:url(" + url + ")";
+                return "background-image:url(" + (url || "/images/css/file_preview.png") + ")";
```
Comment (joozek78) [inline src/Images/wwwroot/Images/elements/so-juicy-filedrop/so-juicy-filedrop.html]: please create a separate method returning style for preview. It's not intuitive to return style for preview when no argument is provided

**Comment by joozek78** [comment]
BTW if you want to reference an issue from a different repository in a commit message you need to include the org name (so starcounter/epicdemo#25)

File: src/Images/ViewModels/ConceptPage.json (hunk 1 above)
Comment (diliger) [inline src/Images/ViewModels/ConceptPage.json]: Fixed.

File: src/Images/ViewModels/ConceptPage.json.cs (hunk 2 above)
Comment (diliger) [inline src/Images/ViewModels/ConceptPage.json.cs]: Fixed.

File: src/Images/ViewModels/ConceptPage.json.cs (hunk 3 above)
Comment (diliger) [inline src/Images/ViewModels/ConceptPage.json.cs]: Fixed for multiple files.

File: src/Images/ViewModels/ConceptPage.json.cs (hunk 4 above)
Comment (diliger) [inline src/Images/ViewModels/ConceptPage.json.cs]: No, Illustration is a read-only property.

File: src/Images/ViewModels/IllustrationPage.json (hunk 5 above)
Comment (diliger) [inline src/Images/ViewModels/IllustrationPage.json]: Fixed.

File: src/Images/wwwroot/Images/elements/so-juicy-filedrop/so-juicy-filedrop.html (hunk 6 above)
Comment (diliger) [inline src/Images/wwwroot/Images/elements/so-juicy-filedrop/so-juicy-filedrop.html]: Done.

**Comment by warpech** [comment]
After this is finished, it should be merged to the **develop** branch instead of **master** branch, for the reasons explained in https://starcounter.gitbooks.io/guidelines/content/branching.html

Please follow this for all of the sample apps

File: src/Images/ViewModels/IllustrationsPage.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
     "Html": "",
     "Uri": "",
-    "$Illustrations": { "Bind": "_Illustrations" },
+    //"$Illustrations": { "Bind": "_Illustrations" },
```
Comment (joozek78) [inline src/Images/ViewModels/IllustrationsPage.json]: this isn't a valid json comment (to be fair, there are no comments in json). Please remove / uncomment

**Comment by joozek78** [comment]
Apart from [this comment in json](https://github.com/StarcounterPrefabs/Images/pull/37/files/b4e976396e964028aac441501ab14699003e810f..00a2832b66b96b981440a63aced885caca8eada3#r75831211) it's all cool

**Comment by joozek78** [comment]
To merge it to develop:
1. close this PR (without merging)
1. open a new one for develop
1. merge and delete source branch

**Pull request closed without merging**